okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -56,7 +56,14 @@ class RealWebSocket(
   private val originalRequest: Request,
   internal val listener: WebSocketListener,
   private val random: Random,
-  private val pingIntervalMillis: Long
+  private val pingIntervalMillis: Long,
+  /**
+   * For clients this is initially null, and will be assigned to the agreed-upon extensions. For
+   * servers it should be the agreed-upon extensions immediately.
+   */
+  private var extensions: WebSocketExtensions?,
```

Comment:
Should this be a non-nullable lateinit?

Reply:
I think this might be more awkward. Although it’s spiritually lateinit, I can’t assign null to that which makes the client constructor awkward.